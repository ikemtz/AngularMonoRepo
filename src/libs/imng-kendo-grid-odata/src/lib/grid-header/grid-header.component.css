.btn-sm {
  height: 30px;
  border-radius: 2px;
  background-color: #f5f5f5;
  border-color: rgba(0, 0, 0, 0.08);
}

.btn-primary {
  background-color: #007bff;
}

.k-icon {
  padding-bottom: 3px;
}
