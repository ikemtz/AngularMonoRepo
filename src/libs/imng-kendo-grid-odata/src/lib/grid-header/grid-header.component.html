<div>
  <button
    *ngIf="entityName"
    type="button"
    title="Add {{ entityName }}"
    primary="true"
    (click)="addItemClicked.emit()"
    class="btn btn-sm btn-primary mx-1"
  >
    <span class="k-icon k-i-plus-circle"></span> Add {{ entityName }}
  </button>
  <button type="button" title="Clear Filters" (click)="clearFiltersClicked.emit()" class="btn btn-sm mx-1">
    <span class="k-icon k-i-filter-clear"></span> Clear Filters
  </button>
  <button
    type="button"
    title="Clear Cache And Reload Data"
    (click)="reloadEntitiesClicked.emit()"
    class="btn btn-sm mx-1"
  >
    <span class="k-icon k-i-reset"></span> Reload Data
  </button>
  <button type="button" title="Export To PDF" kendoGridPDFCommand icon="file-pdf" class="mx-1">Export to PDF</button>
  <button type="button" title="Export To Excel" kendoGridExcelCommand icon="file-excel" class="mx-1">
    Export To Excel
  </button>
  <kendo-grid-column-chooser [allowHideAll]="true" [autoSync]="true" class="mr-5 pr-5"></kendo-grid-column-chooser>
</div>
