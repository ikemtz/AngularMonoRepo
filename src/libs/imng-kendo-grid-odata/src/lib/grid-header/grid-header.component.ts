import { Component, EventEmitter, Input, Output } from '@angular/core';

@Component({
  selector: 'imng-kendo-odata-grid-header',
  templateUrl: './grid-header.component.html',
  styleUrls: ['./grid-header.component.scss'],
})
export class ImngGridHeaderComponent {
  @Input()
  public entityName: string;
  @Output()
  public addItemClicked = new EventEmitter();
  @Output()
  public clearFiltersClicked = new EventEmitter();
  @Output()
  public reloadEntitiesClicked = new EventEmitter();
}
