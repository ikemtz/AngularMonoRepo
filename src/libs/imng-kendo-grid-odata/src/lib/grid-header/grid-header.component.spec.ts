import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { of } from 'rxjs';
import { ImngGridHeaderComponent } from './grid-header.component';

describe('ImngGridHeaderComponent', () => {
  let component: ImngGridHeaderComponent;
  let fixture: ComponentFixture<ImngGridHeaderComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ImngGridHeaderComponent],
      schemas: [NO_ERRORS_SCHEMA],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ImngGridHeaderComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should support hiddenColumns = true', () => {
    component.hasHiddenColumns$ = of(true);
    fixture.detectChanges();
    const element = fixture.nativeElement;
    expect(element.innerHTML).toContain('"text-primary mr-5 pr-5"');
  });

  it('should support hiddenColumns = false', () => {
    component.hasHiddenColumns$ = of(false);
    fixture.detectChanges();
    const element = fixture.nativeElement;
    expect(element.innerHTML).toContain('"mr-5 pr-5"');
  });

  it('should support hideReloadData = true', () => {
    component.hideReloadData = true;
    fixture.detectChanges();
    const element = fixture.nativeElement;
    expect(element.innerHTML).toMatchSnapshot();
  });

  it('should support hideReloadData = false', () => {
    component.hideReloadData = false;
    fixture.detectChanges();
    const element = fixture.nativeElement;
    expect(element.innerHTML).toMatchSnapshot();
  });

  it('should support hideClearFilters = true', () => {
    component.hideClearFilters = true;
    fixture.detectChanges();
    const element = fixture.nativeElement;
    expect(element.innerHTML).toMatchSnapshot();
  });

  it('should support hideClearFilters = false', () => {
    component.hideClearFilters = false;
    fixture.detectChanges();
    const element = fixture.nativeElement;
    expect(element.innerHTML).toMatchSnapshot();
  });
});
