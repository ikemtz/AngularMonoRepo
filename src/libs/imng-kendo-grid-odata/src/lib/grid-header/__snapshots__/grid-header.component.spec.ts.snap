// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImngGridHeaderComponent should support hideClearFilters = false 1`] = `
"<div><!--bindings={}--><button type=\\"button\\" title=\\"Clear Filters\\" class=\\"btn btn-sm mx-1\\"><span class=\\"k-icon k-i-filter-clear\\"></span> Clear Filters </button><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><button type=\\"button\\" title=\\"Clear Cache And Reload Data\\" class=\\"btn btn-sm mx-1\\"><span class=\\"k-icon k-i-reset\\"></span> Reload Data </button><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><button type=\\"button\\" title=\\"Export To PDF\\" kendogridpdfcommand=\\"\\" icon=\\"file-pdf\\" class=\\"mx-1\\">Export to PDF</button><button type=\\"button\\" title=\\"Export To Excel\\" kendogridexcelcommand=\\"\\" icon=\\"file-excel\\" class=\\"mx-1\\"> Export To Excel </button><kendo-grid-column-chooser ng-reflect-ng-class=\\"[object Object]\\" class=\\"mr-2 pr-2\\"></kendo-grid-column-chooser><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--></div>"
`;

exports[`ImngGridHeaderComponent should support hideClearFilters = true 1`] = `
"<div><!--bindings={}--><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--><button type=\\"button\\" title=\\"Clear Cache And Reload Data\\" class=\\"btn btn-sm mx-1\\"><span class=\\"k-icon k-i-reset\\"></span> Reload Data </button><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><button type=\\"button\\" title=\\"Export To PDF\\" kendogridpdfcommand=\\"\\" icon=\\"file-pdf\\" class=\\"mx-1\\">Export to PDF</button><button type=\\"button\\" title=\\"Export To Excel\\" kendogridexcelcommand=\\"\\" icon=\\"file-excel\\" class=\\"mx-1\\"> Export To Excel </button><kendo-grid-column-chooser ng-reflect-ng-class=\\"[object Object]\\" class=\\"mr-2 pr-2\\"></kendo-grid-column-chooser><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--></div>"
`;

exports[`ImngGridHeaderComponent should support hideReloadData = false 1`] = `
"<div><!--bindings={}--><button type=\\"button\\" title=\\"Clear Filters\\" class=\\"btn btn-sm mx-1\\"><span class=\\"k-icon k-i-filter-clear\\"></span> Clear Filters </button><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><button type=\\"button\\" title=\\"Clear Cache And Reload Data\\" class=\\"btn btn-sm mx-1\\"><span class=\\"k-icon k-i-reset\\"></span> Reload Data </button><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><button type=\\"button\\" title=\\"Export To PDF\\" kendogridpdfcommand=\\"\\" icon=\\"file-pdf\\" class=\\"mx-1\\">Export to PDF</button><button type=\\"button\\" title=\\"Export To Excel\\" kendogridexcelcommand=\\"\\" icon=\\"file-excel\\" class=\\"mx-1\\"> Export To Excel </button><kendo-grid-column-chooser ng-reflect-ng-class=\\"[object Object]\\" class=\\"mr-2 pr-2\\"></kendo-grid-column-chooser><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--></div>"
`;

exports[`ImngGridHeaderComponent should support hideReloadData = true 1`] = `
"<div><!--bindings={}--><button type=\\"button\\" title=\\"Clear Filters\\" class=\\"btn btn-sm mx-1\\"><span class=\\"k-icon k-i-filter-clear\\"></span> Clear Filters </button><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--><!--bindings={
  \\"ng-reflect-ng-if\\": \\"false\\"
}--><button type=\\"button\\" title=\\"Export To PDF\\" kendogridpdfcommand=\\"\\" icon=\\"file-pdf\\" class=\\"mx-1\\">Export to PDF</button><button type=\\"button\\" title=\\"Export To Excel\\" kendogridexcelcommand=\\"\\" icon=\\"file-excel\\" class=\\"mx-1\\"> Export To Excel </button><kendo-grid-column-chooser ng-reflect-ng-class=\\"[object Object]\\" class=\\"mr-2 pr-2\\"></kendo-grid-column-chooser><!--bindings={
  \\"ng-reflect-ng-if\\": \\"true\\"
}--></div>"
`;
